File: platform/processing-core/app/schemas/admin/partner_legal.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class PartnerLegalProfileStatusUpdate(BaseModel):
    status: str
    comment: str | None = None


class PartnerLegalPackRequest(BaseModel):
    format: str = "ZIP"


class PartnerLegalPackOut(BaseModel):
    id: str
    partner_id: str
    format: str
    object_key: str
    pack_hash: str
    metadata: dict | None = None
    created_at: datetime
    download_url: str | None = None


class PartnerLegalPackHistoryResponse(BaseModel):
    items: list[PartnerLegalPackOut]


class PartnerLegalProfileAdminOut(BaseModel):
    partner_id: str
    legal_type: str | None = None
    country: str | None = None
    tax_residency: str | None = None
    tax_regime: str | None = None
    vat_applicable: bool | None = None
    vat_rate: float | None = None
    legal_status: str | None = None
    details: dict | None = None
    tax_context: dict | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None

File: frontends/admin-ui/src/api/partnerLegal.ts
import { request } from "./http";

export type PartnerLegalProfileAdmin = {
  partner_id: string;
  legal_type?: string | null;
  country?: string | null;
  tax_residency?: string | null;
  tax_regime?: string | null;
  vat_applicable?: boolean | null;
  vat_rate?: number | null;
  legal_status?: string | null;
  details?: Record<string, unknown> | null;
  tax_context?: Record<string, unknown> | null;
};

export type PartnerLegalPack = {
  id: string;
  partner_id: string;
  format: string;
  object_key: string;
  pack_hash: string;
  metadata?: Record<string, unknown> | null;
  created_at: string;
  download_url?: string | null;
};

export async function fetchPartnerLegalProfile(token: string, partnerId: string) {
  return request<PartnerLegalProfileAdmin>(`/admin/partners/${partnerId}/legal-profile`, { method: "GET" }, token);
}

export async function updatePartnerLegalStatus(
  token: string,
  partnerId: string,
  payload: { status: string; reason: string; correlation_id: string; comment?: string | null },
) {
  return request<PartnerLegalProfileAdmin>(
    `/admin/partners/${partnerId}/legal-profile/status`,
    { method: "POST", body: JSON.stringify(payload) },
    token,
  );
}

export async function generatePartnerLegalPack(token: string, partnerId: string, format = "ZIP") {
  return request<PartnerLegalPack>(
    `/admin/partners/${partnerId}/legal-pack`,
    { method: "POST", body: JSON.stringify({ format }) },
    token,
  );
}

export async function fetchPartnerLegalPackHistory(token: string, partnerId: string) {
  const response = await request<{ items: PartnerLegalPack[] }>(
    `/admin/partners/${partnerId}/legal-pack/history`,
    { method: "GET" },
    token,
  );
  return response.items ?? [];
}
